Upstream changes to ClickEvent breaks compilation

Changes to `ClickEvent` introduced in https://github.com/zed-industries/zed/pull/35075 breaks compilation.
```
error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\title_bar.rs:187:48
    |
187 | ...                   f(&ClickEvent::default(), window, cx);
    |                                      ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\drawer.rs:151:55
    |
151 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\drawer.rs:166:59
    |
166 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\drawer.rs:207:71
    |
207 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:303:56
    |
303 | ...                   if !on_ok(&ClickEvent::default(), window, cx) {
    |                                              ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:308:51
    |
308 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:331:56
    |
331 | ...                   if !on_cancel(&ClickEvent::default(), window, cx) {
    |                                                  ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:335:51
    |
335 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:390:56
    |
390 | ...                   on_cancel(&ClickEvent::default(), window, cx);
    |                                              ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:391:55
    |
391 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:419:64
    |
419 | ...                   on_cancel(&ClickEvent::default(), window, cx);
    |                                              ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:420:63
    |
420 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:430:67
    |
430 | ...                   if on_ok(&ClickEvent::default(), window, cx) {
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:431:71
    |
431 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:468:68
    |
468 | ...                   on_cancel(&ClickEvent::default(), window, cx);
    |                                              ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0599]: no variant or associated item named `default` found for enum `ClickEvent` in the current scope
   --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\modal.rs:469:67
    |
469 | ...                   on_close(&ClickEvent::default(), window, cx);
    |                                             ^^^^^^^ variant or associated item not found in `ClickEvent`

error[E0609]: no field `up` on type `&ClickEvent`
  --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\9a040ea\crates\ui\src\event.rs:13:22
   |
13 |             if event.up.click_count == 2 {
   |                      ^^ unknown field

Some errors have detailed explanations: E0599, E0609.
For more information about an error, try `rustc --explain E0599`.
error: could not compile `gpui-component` (lib) due to 17 previous errors
warning: build failed, waiting for other jobs to finish...
```

@huacnlee Thanks, but this also needs to be fixed.
```
error[E0609]: no field `up` on type `&ClickEvent`
  --> C:\Users\zhe\.cargo\git\checkouts\gpui-component-95ce574d8a0da8b8\0ab497e\crates\ui\src\event.rs:13:22
   |
13 |             if event.up.click_count == 2 {
   |                      ^^ unknown field

```

--- a/crates/ui/src/event.rs
+++ b/crates/ui/src/event.rs
@@ -10,7 +10,7 @@ pub trait InteractiveElementExt: InteractiveElement {
         Self: Sized,
     {
         self.interactivity().on_click(move |event, window, cx| {
-            if event.up.click_count == 2 {
+            if event.click_count() == 2 {
                 listener(event, window, cx);
             }
         });

